server/rosetta/client_offline.go

```diff
@@ -0,0 +1,208 @@
+	if len(request.Operations) > 3 {
+		return nil, crgerrs.ErrInvalidOperation
+	}
+
+	msgs, fee, err := operationsToSdkMsgs(c.ir, request.Operations)
+	if err != nil {
+		return nil, crgerrs.WrapError(crgerrs.ErrInvalidOperation, err.Error())
+	}
+
+	metadata, err := getMetadataFromPayloadReq(request)
+	if err != nil {
+		return nil, crgerrs.WrapError(crgerrs.ErrBadArgument, err.Error())
+	}
+
+	txFactory := tx.Factory{}.WithAccountNumber(metadata.AccountNumber).WithChainID(metadata.ChainID).
+		WithGas(metadata.Gas).WithSequence(metadata.Sequence).WithMemo(metadata.Memo).WithFees(fee.String())
+
+	TxConfig := c.getTxConfig()
+	txFactory = txFactory.WithTxConfig(TxConfig)
+
+	txBldr, err := tx.BuildUnsignedTx(txFactory, msgs...)
+	if err != nil {
+		return nil, err
+	}
+
+	if txFactory.SignMode() == signing.SignMode_SIGN_MODE_UNSPECIFIED {
+		txFactory = txFactory.WithSignMode(signing.SignMode_SIGN_MODE_LEGACY_AMINO_JSON)
+	}
+
+	signerData := authsigning.SignerData{
+		ChainID:       txFactory.ChainID(),
+		AccountNumber: txFactory.AccountNumber(),
+		Sequence:      txFactory.Sequence(),
+	}
+
+	signBytes, err := TxConfig.SignModeHandler().GetSignBytes(txFactory.SignMode(), signerData, txBldr.GetTx())
+	if err != nil {
+		return nil, err
+	}
+
+	txBytes, err := TxConfig.TxEncoder()(txBldr.GetTx())
+	if err != nil {
+		return nil, err
+	}
+
+	accIdentifiers := getAccountIdentifiersByMsgs(msgs)
+
+	payloads := make([]*types.SigningPayload, len(accIdentifiers))
+	for i, accID := range accIdentifiers {
+		payloads[i] = &types.SigningPayload{
+			AccountIdentifier: accID,
+			Bytes:             crypto.Sha256(signBytes),
+			SignatureType:     "ecdsa",
+		}
+	}
+
+	return &types.ConstructionPayloadsResponse{
+		UnsignedTransaction: hex.EncodeToString(txBytes),
+		Payloads:            payloads,
+	}, nil
+}
+
+func getAccountIdentifiersByMsgs(msgs []sdk.Msg) []*types.AccountIdentifier {
+	var accIdentifiers []*types.AccountIdentifier
+	for _, msg := range msgs {
+		for _, signer := range msg.GetSigners() {
+			accIdentifiers = append(accIdentifiers, &types.AccountIdentifier{Address: signer.String()})
+		}
+	}
+
+	return accIdentifiers
+}
+
+func (c *Client) PreprocessOperationsToOptions(_ context.Context, req *types.ConstructionPreprocessRequest) (options map[string]interface{}, err error) {
+	operations := req.Operations
+	if len(operations) > 3 {
+		return nil, crgerrs.WrapError(crgerrs.ErrBadArgument, "invalid number of operations")
+	}
+
+	msgs, err := ConvertOpsToMsgs(c.ir, operations)
+	if err != nil {
+		return nil, crgerrs.WrapError(crgerrs.ErrInvalidOperation, err.Error())
+	}
+
+	if len(msgs) < 1 || len(msgs[0].GetSigners()) < 1 {
+		return nil, crgerrs.WrapError(crgerrs.ErrInvalidOperation, "invalid msgs from operations")
+	}
+
+	memo, ok := req.Metadata["memo"]
+	if !ok {
+		memo = ""
+	}
+
+	defaultGas := float64(200000)
```

Comment:
why?

Reply:
@sahith-narahari

Reply:
this was to keep it aligned with the default value of gas being used in sdk